### PR TITLE
feat(fileName): Add hash support

### PR DESCRIPTION
**Pull request by obartra**
Adds `[hash]` and `[hash:size]` support

### CODE REVIEW

**Comment by obartra** [comment]
the build is failing because master is failing (missing `extract-text-plugin`)

**Comment by obartra** [comment]
I don't think this is working as intended, the sprite itself is saved with the right name but the resources using it point to an incorrect path. I'm not too familiar with webpack, @kisenka do you have any suggestions?

**Comment by kisenka** [comment]
@obartra 
> the sprite itself is saved with the right name but the resources using it point to an incorrect path

Could you please provide an example of such resources and it's final paths.

File: lib/utils.js [hunk 1]
```diff
@@ -266,11 +272,14 @@ module.exports.getModuleChunk = getModuleChunk;
  * @param {Object} options
  * @param {string} options.chunkName
  * @param {string} options.context
+ * @param {string} options.fullHash
```
Comment (obartra) [inline lib/utils.js]: this is webpack's `fullHash` though, happy to rename it but I figured I would keep it consistent

File: lib/utils.js [hunk 2]
```diff
@@ -266,11 +272,14 @@ module.exports.getModuleChunk = getModuleChunk;
...
  * @param {string} options.content
  * @return {string}
  */
 function interpolateSpriteFilename(filename, options) {
-  return filename.replace(/\[chunkname\]/g, options.chunkName);
+  return filename
+    .replace(/\[hash(:\d)?]/g, getSizedMatch.bind(this, options.fullHash))
```
Comment (obartra) [inline lib/utils.js]: can we use loader-utils directly to interpolate the name?

File: lib/utils.js [hunk 3]
```diff
@@ -258,6 +258,12 @@ function getModuleChunk(module, modules) {
   return null;
 }
```
Comment (obartra) [inline lib/utils.js]: Oops totally missed this, will update.

May I suggest [inch-ci](https://inch-ci.org/) and a script like [this](https://github.com/obartra/ssim/blob/master/ci/doc.sh) to automate JSDoc validation as part of CI?

File: lib/utils.js (hunk 2 above)
Comment (kisenka) [inline lib/utils.js]: yes, but only for [hash] pattern, all filename related replacements has no sense

**Comment by obartra** [comment]
With the current implementation the sprite is saved with the full hash, something like `test-[hash:2].svg` could become `test-ab.svg` but the svg use `/test-.svg#iconname` as xlink:href instead (so no hash).

Just by adding a bunch of console log statements I see in some cases fullHash is defined but not on others. I'm guessing some of the events svg-sprite-loader listens to occur before the hash is created

File: lib/utils.js (hunk 3 above)
Comment (kisenka) [inline lib/utils.js]: Cool! Will setup it asap

**Comment by kisenka** [comment]
Ok, leave it to me

**Comment by obartra** [comment]
Is this safe to close?

**Comment by kisenka** [comment]
Yep

**Pull request closed without merging**